learning-another-language/classes.md

```diff
@@ -0,0 +1,361 @@
+
+A zine is like a small, self-published book. They have titles and authors (or contributors, illustrators, editors, writers, etc.), pages, and content.
+
+We chose the example class `Zine` because we were bored of teaching `Book`.
+
+## Syntax
+
+In Python, defining a class and then creating an instance of one looks like this:
+
+```python
+class Zine:
+  def __init__(self,title)
+    self.title = title
+
+zine = Zine('So you want to be a pastry chef')
+```
+
+In JavaScript, defining a class and then creating an instance of one looks like this:
+
+```javascript
+class Zine {
+  constructor(title) {
+    this.title = title;
+  }
+}
+
+const zine = new Zine('So you want to be a pastry chef');
+```
+
+### Try it
+Let's make our first instances of the `Zine` class! Open up the node console and put in the definition for the `Zine` class (typing the code from above into the node console or copying/pasting it into the node console will both work).
+
+Then, make many instances of the `Zine` class. Make an instance of `Zine` with the following titles:
+
+1. So you want to be a pastry chef
+1. Bite Size Command Line!
+1. Linux debugging tools you'll love
+
+You should get input that looks like this:
+
+```bash
+const chefZine = new Zine('So you want to be a pastry chef');
+> undefined
+chefZine
+> Zine { title: 'So you want to be a pastry chef' }
+const commandLineZine = new Zine('Bite Size Command Line!');
+> undefined
+commandLineZine
+> Zine { title: 'Bite Size Command Line!' }
+const linuxZine = new Zine("Linux debugging tools you'll love");
+> undefined
+linuxZine
+> Zine { title: "Linux debugging tools you'll love" }
+```
+
+Congratulations! You made your first class in JavaScript!
+
+How do you read attributes?
+
+In JavaScript, instances of classes aren't anything special, you just get objects. So, reading an attribute from an instance of a class is the same as reading an attribute from any other object with the same attributes.
+
+To read or write the `title` we set in the constructor, we can use both dot notation and subscript notation (square brackets):
+
+```javascript
+// You can read/write with 'dot notation'
+console.log(chefZine.title);
+> So you want to be a pastry chef
+
+chefZine.title = 'A much better title';
+console.log(chefZine.title);
+> A much better title
+
+// Or with bracket notation
+console.log(chefZine['title']);
+> A much better title
+
+chefZine["title"] = 'Frankly, this title is a little mediocre'
+console.log(chefZine['title']);
+> Frankly, this title is a little mediocre
+```
+
+**Experiment:** What happens if you try to read or write an attribute that isn't set in the constructor?
```

Comment:
This might be a good time for a `<details>` tag

```suggestion
<details style="max-width: 700px; margin: auto;">
  <summary>
  **Experiment:** What happens if you try to read or write an attribute that isn't set in the constructor?
  </summary>
  
  Any attribute not defined in the constructor is `undefined`.
</details>
```